packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -142,109 +94,493 @@ describe("Decentraland", () => {
+            contractAddress,
+            contractABI,
+            functionName,
+            argsArray,
+            ethersAmount
+          );
+
+          const onFailed = message => {
+            const { data } = message;
+            const { args } = data;
+            const { txHash } = args;
+          };
+          gnosisSafe.once("ExecutionFailed", onFailed);
+
+          await executeOrderAction.waitForNonceToUpdate();
+
+          await mineBlocks(provider, 1);
+
+          await executeOrderAction.waitForNonceToUpdate();
+
+          await confirmBalances(land, [GNOSIS_SAFE_ADDRESS], [1]);
+        });
+      });
 
-        await confirmBalances(landContract, [ephemeralAddress], [1]);
+      // Allowence-of-allowence doesn't work
+      // See more: https://github.com/tasitlabs/TasitSDK/issues/273
+      describe.skip("Using an ephemeral wallet allowed to spend Gnosis Safe wallets' funds", () => {
+        beforeEach("onboarding", async () => {
+          // Funding ephemeral account with some ethers to pay for gas
+          // TODO: ephemeralWallet should broadcast this action
+          const toAddress = ephemeralAddress;
+          gnosisSafe.setSigners([gnosisSafeOwner]);
+          gnosisSafe.setWallet(gnosisSafeOwner);
+          const transferEthersAction = gnosisSafe.transferEther(
+            toAddress,
+            SMALL_AMOUNT
+          );
+          await transferEthersAction.waitForNonceToUpdate();
+          await confirmEtherBalances(provider, [toAddress], [SMALL_AMOUNT]);
+
+          // Gnosis Safe should approve ephemeral account to spend its MANA Tokens
+          // TODO: ephemeralWallet should broadcast this action
+          const contractAddress = mana.getAddress();
+          const contractABI = mana.getABI();
+          const functionName = "approve";
+          const spender = ephemeralAddress;
+          const argsArray = [spender, manaAmountForShopping];
+          gnosisSafe.setSigners([gnosisSafeOwner]);
+          gnosisSafe.setWallet(gnosisSafeOwner);
+          const ephemeralApprovalAction = gnosisSafe.customContractAction(
+            contractAddress,
+            contractABI,
+            functionName,
+            argsArray
+          );
+          await ephemeralApprovalAction.waitForNonceToUpdate();
+          const owner = GNOSIS_SAFE_ADDRESS;
+          const ephemeralAllowance = await mana.allowance(owner, spender);
+          expect(`${ephemeralAllowance}`).to.equal(`${manaAmountForShopping}`);
+
+          // Ephemeral account approves Marketplace to spend their mana tokens (actually tokens from Gnosis Safe).
+          mana.setWallet(ephemeralWallet);
+          const marketplaceApprovalAction = mana.approve(
+            MARKETPLACE_ADDRESS,
+            manaAmountForShopping,
+            gasParams
+          );
+          await marketplaceApprovalAction.waitForNonceToUpdate();
+          const marketplaceAllowance = await mana.allowance(
```

Comment:
To be very explicit, maybe do an expectation on the Safe's allowance for the ephemeral account, the Safe's allowance for the marketplace, and the ephemeral account's allowance for the marketplace?